### PR TITLE
Fix SonarCloud Issues reported on Dockerfile

### PR DESCRIPTION
**Pull request by DrizzlyOwl**
## Changes

- Variables are properly escaped
- No vendor recommended packages are installed via `apt-get`
- References to version pins are made consistent
- Create a non-root user and assign it permission to manage the app home dir
- Run the Docker container as the non-root user
- Ensure references to `curl` are enforcing TLS connections and HTTPS protocol and limiting redirects where possible to prevent domain highjacking
- Clean up apt cache after installing packages

## Checklist

- [x] Attach this pull request to the appropriate card in DevOps.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Review by mec — COMMENTED** [review]
Thanks for all the effort on this! 👍 I am happy to go, but I think @lozette should have site of these change as well and then approve.

File: Dockerfile [hunk 1]
```diff
@@ -107,6 +117,7 @@ RUN \
   else \
     yarn install --frozen-lockfile; \
   fi
+USER root
```
Comment (lozette) [inline Dockerfile:121]: Question bc I don't understand it: why do we switch back to root here?

Comment (mec) [reply Dockerfile:121]: I had assumed that eveything after has to run as root, up until the next `USER` - but @DrizzlyOwl should advise?

Comment (DrizzlyOwl) [reply Dockerfile:121]: We might be able to remove this line actually

**Review by lozette — APPROVED** [review]

**Pull request merged**